stripe/src/test/java/com/stripe/android/paymentsheet/PaymentOptionsViewModelTest.kt

```diff
@@ -81,6 +91,80 @@ class PaymentOptionsViewModelTest {
         )
     }
 
+    @Test
+    fun `resolveTransitionTarget no new card`() {
+
+        val viewModel = PaymentOptionsViewModel(
+            args = args.copy(newCard = null),
+            prefsRepository = FakePrefsRepository(),
+            eventReporter = eventReporter
+        )
+
+        var transitionTarget: TransitionTarget? = null
+        viewModel.transition.observeForever {
+            transitionTarget = it
+        }
+
+        // no customer, no new card, no paymentMethods
+        val fragmentConfig = FragmentConfigFixtures.DEFAULT
+        viewModel.resolveTransitionTarget(fragmentConfig)
+
+        assertThat(transitionTarget).isNull()
+    }
+
+    @Test
+    fun `resolveTransitionTarget new card saved`() {
+
+        val viewModel = PaymentOptionsViewModel(
+            args = args.copy(
+                newCard = newCard.copy(
+                    shouldSavePaymentMethod = true
+                )
+            ),
+            prefsRepository = FakePrefsRepository(),
+            eventReporter = eventReporter
+        )
+
+        var transitionTarget: TransitionTarget? = null
+        viewModel.transition.observeForever {
+            transitionTarget = it
+        }
+
+        val fragmentConfig = FragmentConfigFixtures.DEFAULT
+        viewModel.resolveTransitionTarget(fragmentConfig)
+
+        assertThat(transitionTarget).isNull()
+    }
+
+    @Test
+    fun `resolveTransitionTarget new card NOT saved`() {
+
+        val viewModel = PaymentOptionsViewModel(
+            args = args.copy(
+                newCard = newCard.copy(
+                    shouldSavePaymentMethod = false
+                )
+            ),
+            prefsRepository = FakePrefsRepository(),
+            eventReporter = eventReporter
+        )
+
+        var transitionTarget: TransitionTarget? = null
```

Comment:
You can make `transitionTarget` a `MutableList<TransitionTarget>` and then you can get rid of `numTimesCalled`